sktime/classifiers/shapelet_based/mrseql/mrseql.pyx

```diff
@@ -0,0 +1,439 @@
+                first_char += self.sfa.alphabet_size
+        return sfa_str
+
+###########################################################################
+
+
+#########################SEQL wrapper#########################
+
+
+cdef extern from "seql.h":
+    cdef cppclass SEQL:
+        SEQL()
+        void learn(vector[string] &, vector[double] &)
+        double brute_classify(string , double)
+        void print_model(int)
+        vector[string] get_sequence_features(bool)
+        vector[double] get_coefficients(bool)
+
+# class BaseLearner:
+#     def learn(self, sequences, labels):
+#         raise NotImplementedError('this is an abstract method')
+
+#     def classify(self, sequence):
+#         raise NotImplementedError('this is an abstract method')    
+
+#     def get_sequence_features(self, only_positive):
+#         raise NotImplementedError('this is an abstract method')
+
+#     def get_coefficients(self, only_positive):
+#         raise NotImplementedError('this is an abstract method')
+
+
+cdef class PySEQL:
+    cdef SEQL *thisptr
+
+    def __cinit__(self):
+        self.thisptr = new SEQL()
+    def __dealloc__(self):
+        del self.thisptr
+
+    def learn(self, vector[string] sequences, vector[double] labels):
+        self.thisptr.learn(sequences, labels)
+
+    def classify(self, string sequence):
+        scr = self.thisptr.brute_classify(sequence, 0.0)
+        return np.array([-scr,scr]) # keep consistent with multiclass case
+
+    def print_model(self):
+        self.thisptr.print_model(100)
+
+    def get_sequence_features(self, bool only_positive = False):
+        return self.thisptr.get_sequence_features(only_positive)
+
+    def get_coefficients(self, bool only_positive = False):
+        return self.thisptr.get_coefficients(only_positive)
+
+class OVASEQL:
+    def __init__(self, unique_labels):
+        self.labels_ = unique_labels
+        self.models = []
+
+    def learn(self, sequences, labels):
+        for l in self.labels_:
+            tmp_labels = [1 if c == l else - 1 for c in labels]
+            m = PySEQL()
+            m.learn(sequences, tmp_labels)
+            self.models.append(m)
+
+    def classify(self, string sequence):
+        scr = []
+        for m in self.models:
+            scr.append(m.classify(sequence)[1])
+        return np.array(scr)
+
+    def get_sequence_features(self, bool only_positive = False):
+        sqs = []
+        for m in self.models:
+            sqs.extend(m.get_sequence_features(True))
+        return sqs
+
+    def get_coefficients(self, bool only_positive = False):
+        coefs = []
+        for m in self.models:
+            coefs.extend(m.get_coefficients(True))
+        return coefs
+
+
+###########################################################################
+
+
+######################### Mr-SEQL (main class) #########################
+''' Time Series Classification with multiple symbolic representations and SEQL (Mr-SEQL)
```

Comment:
Docstring should go inside the class, I know we haven't been very good at it ourselves, but please try to adhere to scikit-learn's code/docstring formatting.